src/cascadia/TerminalSettingsModel/VsDevCmdGenerator.cpp

```diff
@@ -45,7 +45,7 @@ std::wstring VsDevCmdGenerator::GetProfileCommandLine(const VsSetupConfiguration
     // The "-startdir" parameter will prevent "vsdevcmd" from automatically
     // setting the shell path so the path in the profile will be used instead.
 #if defined(_M_ARM64)
-    commandLine.append(LR"(" -startdir=none -arch=arm64 -host_arch=x64)");
+    commandLine.append(LR"(" -startdir=none -arch=arm64 -host_arch=arm64)");

```

Comment:
Alright, I hate it but... it [looks like](https://learn.microsoft.com/en-us/visualstudio/install/visual-studio-on-arm-devices?view=vs-2022) 17.4 is the version that added arm64 native hosting.

Would you mind adding a check like we have on line R85 in the powershell dev shell generator?

https://github.com/microsoft/terminal/blob/1142b6c0928a9a9127c679ffb081ae3f9b007da5/src/cascadia/TerminalSettingsModel/VsDevShellGenerator.cpp#L85